sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_store.hpp

```diff
@@ -0,0 +1,178 @@
+//==------- block_store.hpp - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-------------------------------------------------------------------===//
+
+#include "common.hpp"
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+template <typename T, uint16_t N, bool UseMask, bool UseProperties,
+          typename StorePropertiesT>
+bool testUSM(queue Q, uint32_t Groups, uint32_t Threads,
+             StorePropertiesT StoreProperties) {
+
+  uint16_t Size = Groups * Threads * N;
+  using Tuint = sycl::_V1::ext::intel::esimd::detail::uint_type_t<sizeof(T)>;
+
+  std::cout << "USM case: T=" << esimd_test::type_name<T>() << ",N=" << N
+            << ",UseMask=" << UseMask << ",UseProperties=" << UseProperties
+            << std::endl;
+
+  sycl::range<1> GlobalRange{Groups};
+  sycl::range<1> LocalRange{Threads};
+  sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+  constexpr size_t Alignment = getAlignment<T, N, UseMask>(StoreProperties);
+  T *Out = sycl::aligned_alloc_shared<T>(Alignment, Size, Q);
+  T Out_val = esimd_test::getRandomValue<T>();
+  for (int i = 0; i < Size; i++)
+    Out[i] = Out_val;
+
+  try {
+    Q.submit([&](handler &cgh) {
+       cgh.parallel_for(Range, [=](sycl::nd_item<1> ndi) SYCL_ESIMD_KERNEL {
+         uint16_t GlobalID = ndi.get_global_id(0);
+         uint32_t ElemOff = GlobalID * N;
+         //  TODO: these 2 lines work-around the problem with scalar
+         //  conversions to bfloat16. It could be just: "simd<T, N>
+         //  PassThru(ElemOffset, 1);"
+         simd<uint32_t, N> PassThruInt(ElemOff, 1);
+         simd<T, N> Vals = PassThruInt;
+         if constexpr (UseMask) {
+           simd_mask<1> Mask = (GlobalID + 1) % 1;
```

Comment:
Hi Nick,
I just found an error in block_load.hpp, and will fix it soon and it seems you copy-pasted it to this block_store.hpp.
The code that was supposed to be here and in few other places is: `simd_mask<1> Mask = (GlobalID + 1) & 0x1;`

Reply:
(Val % 1) always gives 0.

Reply:
Will do this now, thanks for the heads up

Reply:
Luckily it passes, I'm making a PR now